sdk/storage/test-resources.json

```diff
@@ -366,74 +370,106 @@
-    },
-    "PREMIUM_STORAGE_ACCOUNT_NAME": {
-      "type": "string",
-      "value": "[variables('premiumAccountName')]"
-    },
-    "PREMIUM_STORAGE_ACCOUNT_KEY": {
-      "type": "string",
-      "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('premiumAccountName')), variables('mgmtApiVersion')).keys[0].value]"
-    },
-    "DATALAKE_STORAGE_ACCOUNT_NAME": {
-      "type": "string",
-      "value": "[variables('dataLakeAccountName')]"
-    },
-    "DATALAKE_STORAGE_ACCOUNT_KEY": {
-      "type": "string",
-      "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('dataLakeAccountName')), variables('mgmtApiVersion')).keys[0].value]"
-    },
-    "SOFT_DELETE_ACCOUNT_NAME": {
-      "type": "string",
-      "value": "[variables('softDeleteAccountName')]"
-    },
-    "SOFT_DELETE_ACCOUNT_KEY": {
-      "type": "string",
-      "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('softDeleteAccountName')), variables('mgmtApiVersion')).keys[0].value]"
-    },
-    "PREMIUM_FILE_STORAGE_ACCOUNT_NAME": {
-      "type": "string",
-      "value": "[variables('premiumFileAccountName')]"
-    },
-    "PREMIUM_FILE_STORAGE_ACCOUNT_KEY": {
-      "type": "string",
-      "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('premiumFileAccountName')), variables('mgmtApiVersion')).keys[0].value]"
-    },
-    "AZUREWEBJOBSSTORAGE": {
-      "type": "string",
-      "value": "[concat('DefaultEndpointsProtocol=https;AccountName=', variables('webjobsPrimaryAccountName'), ';AccountKey=', listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('webjobsPrimaryAccountName')), variables('mgmtApiVersion')).keys[0].value,';EndpointSuffix=core.windows.net')]"
-    },
-    "AZUREWEBJOBSSECONDARYSTORAGE": {
-      "type": "string",
-      "value": "[concat('DefaultEndpointsProtocol=https;AccountName=', variables('webjobsSecondaryAccountName'), ';AccountKey=', listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('webjobsSecondaryAccountName')), variables('mgmtApiVersion')).keys[0].value,';EndpointSuffix=core.windows.net')]"
-    },
-    "KEYVAULT_URI":{
-      "type": "string",
-      "value": "[variables('azureKeyVaultUrl')]"
+    "outputs": {
+        "PRIMARY_STORAGE_ACCOUNT_NAME": {
+            "type": "string",
+            "value": "[variables('primaryAccountName')]"
+        },
+        "PRIMARY_STORAGE_ACCOUNT_KEY": {
+            "type": "string",
+            "value": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', variables('primaryAccountName')), variables('mgmtApiVersion')).keys[0].value]"
+        },
+        "PRIMARY_STORAGE_ACCOUNT_ZONE": {
+            "type": "string",
+            "value": "[first(skip(split(reference(resourceId('Microsoft.Storage/storageAccounts', variables('primaryAccountName')), variables('mgmtApiVersion')).primaryEndpoints.blob, '.'),1))]"
```

Comment:
This was a roundabout way to grab the value from the URL that I wanted, if there's a not-as-complicated way, let me know.

Reply:
Could you extract a function to encapsulate this logic and avoid repetition . see https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-user-defined-functions